extensions/wikia/TemplateDraft/TemplateConverter.class.php

```diff
@@ -102,4 +110,30 @@ public function createDataTag( $source, $label = '' ) {
 		}
 		return "\t<data source=\"{$source}\"><label>{$label}</label></data>\n";
 	}
+
+	public function generatePreviewSection( $variables ) {
+		$return = "== Usage & preview ==\n";
+		$return .= "<noinclude>\n";
+
+		$preview = "{{" . $this->title->getText() . "\n";
+		$docs = $preview;
+
+		foreach ( $variables as $var ) {
+			$preview .= "|{$var}=This is a test\n";
+			$docs .= "|{$var}=\n";
+		}
+
+		$preview .= "}}\n";
+		$docs .= "}}\n";
+
+		$return = "<noinclude>\n== Usage & preview ==\n";
+
+		$return .= "Type in this:\n<pre>\n$docs</pre>\nto see this:\n$preview\n";
+
+		$return .= "[{{fullurl:PAGENAME}}?action=purge Click here to refresh the preview above]\n";
+
+		$return .= "</noinclude>\n";
+
+		return $return;
```

Comment:
I would say in this case this is not needed because the converter's logic operates purely on the infobox code and is a "gluing" class that puts that code together in a different form. There is no logic indepentent from view in this case (at least not as it's currently built).